### PR TITLE
Blog App: validations and model specs

### PR DESCRIPTION
**Pull request by didierganthier**
## Checklist
- [x] I have read this repository's contribution guidelines.
- [x] My pull request has a descriptive title (not a vague title like Update index.MD)
- [x] My pull request targets the main branch of this [project](https://github.com/didierganthier/my_blog).
- [x] I added relevant documentation for all the files I changed

## Summary
This PR does the following:
- Use validations for models.
- Write unit tests for models.

 ## What changes have been made?
- [x] Add the following validations:
- [x] For the User model:
   - The name must not be blank.
   - PostsCounter must be an integer greater than or equal to zero.
- [x] For the Post model:
   - The title must not be blank.
   - The title must not exceed 250 characters.
- [x] CommentsCounter must be an integer greater than or equal to zero.
- [x] LikesCounter must be an integer greater than or equal to zero.
- [x] Add unit specs for all of your models' methods and validations.

## Author
- @didierganthier

### CODE REVIEW

File: app/models/like.rb [hunk 1]
```diff
@@ -1,10 +1,8 @@
 class Like < ApplicationRecord
-  belongs_to :author, class_name: 'User'
-  belongs_to :post
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  belongs_to :post, class_name: 'Post', foreign_key: 'post_id'
```
Comment (MahmoudBakr23) [inline app/models/like.rb:3]: **Note**
- As long as the association is directed for a model with its default name, so the class name, and foreign key attract are not necessary. just `belongs to post`is fine.

File: app/models/like.rb [hunk 2]
```diff
@@ -1,10 +1,8 @@
...
 
-  after_save :update_posts_likes_counter
-
-  private
+  after_save :update_likes_counter
```
Comment (MahmoudBakr23) [inline app/models/like.rb:5]: **Highlights**
- Well done using the rails helper method after save

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @didierganthier 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

**Highlights**
- Well done creating the models' associations and validations
- Great job with installing RSpec and writing test cases for each model
- Descriptive pr and readme
- Correct Linters

**Optional**
_Please see the note down below._

Cheers and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/like.rb (hunk 1 above)
Comment (didierganthier) [reply app/models/like.rb:3]: Thanks for pointing that out 👍🏽 you're right

**Pull request merged**